mesonbuild/compilers/compilers.py

```diff
@@ -113,6 +113,13 @@ def is_library(fname):
                           'release': ['-O3'],
                           'minsize': ['-Os', '-g']}
 
+arm_buildtype_args = {'plain': [],
+                      'debug': ['-O0', '--debug'],
+                      'debugoptimized': ['-O1', '--debug'],
+                      'release': ['-O3', '-Otime'],
+                      'minsize': ['-O3', '-Ospace'],
```

Comment:
Similarly should this be only `-Ospace`? Every other compiler I know works so that only the last optimization type flag has any effect.

Reply:
Reading through the ARM documentation and help menu there are 2 options 1 being the level of optimizations and 2 being the type of optimization. Hence we have 1 option each for level and type